### ISSUE TITLE
erf optimizations

### ISSUE THREAD

**Issue by gajjanag** [JuliaLang/openlibm#109]
I don't know if this is of interest to openlibm, but recently I did some work in FFmpeg to get in an erf implementation (on broken libms as a fallback) that was fast (~1.5x GNU/openlibm) and reasonably accurate:
https://github.com/FFmpeg/FFmpeg/commit/dd68cde28a44bbf5307d29ee6cb8ebd14985dea5.
Please note that FFmpeg can afford to be slightly more lax wrt accuracy, since it tends to weigh performance more heavily and also because it needs to work with all kinds of upstream libm's so it can't rely on "correctly rounded" functions. Nevertheless, as seen in the message, it is reasonably accurate, and in some cases is correctly rounded when GNU/openlibm are not.

In summary, if people here think this is promising, and that one can use this approach with some possible tweaks in case accuracy is of interest, I can work further on this and come up with something concrete.

**Comment by ViralBShah** [comment]
I doubt we want the default version to be lax with accuracy than what we have right now. Could you characterize that a bit more. If it is clearly superior and faster, then it is an easy decision.

Cc: @simonbyrne

**Comment by simonbyrne** [comment]
I agree, it looks like it could be worth utilising. The Boost licence is fairly permissive, so I don't think that should be a problem.

It doesn't look like they've done any proper error analysis, but there are some empirical results here:
http://www.boost.org/doc/libs/1_60_0/libs/math/doc/html/math_toolkit/sf_erf/error_function.html#math_toolkit.sf_erf.error_function.accuracy

**Comment by gajjanag** [comment]
I tried some simple strategies for tightening the errors, but my hands were generally tied as long double can't be used within FFmpeg (using the 64 bit version from Boost, fma's for the polynomial, etc). The summary is: it still has issues on some inputs, so if one cares about worst case error, it is inferior to the current implementation.

**Comment by simonbyrne** [comment]
We don't want to depend on `long double`s, they're way too slow on modern processors. It would be good to exploit FMAs, but this is a library-wide issue.

**Comment by ViralBShah** [comment]
In general, on accuracy, since this is expected to be a drop in replacement for a libm, I would prefer not to have an implementation that is less accurate than most libms.

**Comment by StefanKarpinski** [comment]
It might be interesting to have alternative versions of some function that sacrifice accuracy for speed.

**Comment by ViralBShah** [comment]
Pretty tough to maintain if the codebase starts diverging from the msun codebase. Would be nice to see a PR, with that caveat.
